src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -177,6 +177,22 @@ internal static bool IsInbox
 
 #endif
 
+        /// <summary>
+        /// True if it is an instance of PowerShell Core
+        /// </summary>
+        public static bool IsPowerShellCore
+        {
+            get
+            {
+#if !CORECLR
+                return false;
+#else
```

Comment:
Handling general code changes as part of #4357 is beyond the scope of this PR and should be handled in a separate one. For example, the STA changes should be bundled in that PR.